docs/rfcs/0020-rotate-hash-peppers.md

```diff
@@ -0,0 +1,379 @@
+# RFC Proposal
+[RFC Proposal]: #rfc-proposal
+
+* RFC Proposal ID: `0020-rotate-hash-peppers`
+* Start Date: 2023-08-21
+* RFC PR: [beneficiary-fhir-data/rfcs#0020](https://github.com/CMSgov/beneficiary-fhir-data/pull/1642)
+* JIRA Ticket(s):
+    * [BFD-1110](https://jira.cms.gov/browse/BFD-1110)
+
+BFD maintains hashed MBI and hashed HICN values to facilitate FHIR patient search for a beneficiary; every beneficiary's hashed values are persisted in the BFD database and can handle both current as well as historical MBI and HICN values. 
+
+The hashing algorithm, and all criteria necessary to create a matching hashed value, is shared with peering partners; this allows them to construct obfuscated MBI or HICN hash values that can then be safely used within an HTTP GET URL parameter(s) accessing the BFD patient search endpoint.
+
+This RFC is therefore intended to explore solutions that lend themselves to providing continued access to BFD patient search functionality with minimal to zero downtime due to either diminished access to BFD services or imposing undue work required of BFD clients (hereafter referred to as Peering Partners or PPs).
+
+## Status
+[Status]: #status
+
+* Status: Proposed <!-- (Proposed/Approved/Rejected/Implemented) -->
+* Implementation JIRA Ticket(s):
+    * [BFD-1110](https://jira.cms.gov/browse/BFD-1110)
+
+## Table of Contents
+[Table of Contents]: #table-of-contents
+
+* [RFC Proposal](#rfc-proposal)
+* [Status](#status)
+* [Table of Contents](#table-of-contents)
+* [Motivation](#motivation)
+* [Definitions](#definitions)
+* [Background](#background)
+  * [Current Patient Search](#current-patient-search)
+  * [BFD Hash Algorithm](#bfd-hash-algorithm)
+  * [Much Ado About Nothing?](#much-ado-about-nothing)
+* [Exploration of Solutions](#exploration-of-solutions)
+  * [Status Quo](#status-quo)
+  * [Hot Stand-by : Option 1](#hot-standby-1)
+  * [Hot Stand-by : Option 2](#hot-standby-2)
+  * [HTTP POST](#http-post)
+  * [Asymmetric Crypto](#asymmetric-crypto)
+* [Proposed Solution: Detailed Design](#proposed-solution-detailed-design)
+  * [Proposed Solution: Unresolved Questions](#proposed-solution-unresolved-questions)
+  * [Proposed Solution: Drawbacks](#proposed-solution-drawbacks)
+  * [Proposed Solution: Notable Alternatives](#proposed-solution-notable-alternatives)
+* [Prior Art](#prior-art)
```

Comment:
I'd like to see a description of the solution from the PACA code base that is already set up to handle old and new hashes in queries and includes code for computing and caching hashed MBIs on the fly.  It could be adapted to adjudicated data as well.  Maybe just inserted as another option here or described in the prior art section.  It's not necessarily superior to any of the options in here but it deserves to be described given it is directly related to the topic.

Reply:
Brian. Would the following in the Prior Art be appropriate?

RDA (PACA) implemented a hashed MBI cache table to their database schema which handles current and previous (old) MBI hash value(s) within their patient lookup queries. If a client requesting pre-adjudicated claims data passes in an old hashed value, and that value exists in the cache table, the query lookup succeeds **AND** the patient's current MBI hash value is added to the cache table if it does not already exist there. In the case of a hash algorithm change, no extra work would be necessary to maintain the cache, as it dynamically maintains itself. Should the adoption of HTTP POST be deferred, this cache table strategy could be adopted for adjudicated claims.

Reply:
I have added a description of how the `mbi_cache` table works as an inline comment.